quixstreams/platforms/quix/api.py

```diff
@@ -61,16 +66,40 @@ def request(self, method, url, **kwargs):
     @property
     def default_workspace_id(self) -> str:
         if not self._default_workspace_id:
-            raise self.UndefinedQuixWorkspaceId("You must provide a Quix Workspace ID")
+            raise UndefinedQuixWorkspaceId(
+                f"A Quix Cloud Workspace ID is required; "
+                f"set with environment variable {QUIX_ENVIRONMENT.WORKSPACE_ID}"
+            )
         return self._default_workspace_id
 
     @default_workspace_id.setter
     def default_workspace_id(self, value):
         self._default_workspace_id = value
 
+    def _response_handler(self, r: requests.Response, *args, **kwargs):
+        """
+        Custom callback/hook that is called after receiving a request.Response
+
+        Catches non-200's and passes both the original exception and the Response body.
+
+        Note: *args and **kwargs expected for hook
+        """
+        try:
+            r.raise_for_status()
+        except requests.exceptions.RequestException as e:
```

Comment:
As mentioned previously, let's catch only `HTTPError` here, not `RequestException`.
Only `HTTPError` exceptions are raised in `raise_for_status()`